src/components/Reports/GoalCalculator/SummaryReport/Steps/PresentingYourGoalStep/GoalApplicationButtonGroup.test.tsx

```diff
@@ -0,0 +1,112 @@
+import { render } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import { GoalCalculatorTestWrapper } from '../../../GoalCalculatorTestWrapper';
+import { GoalApplicationButtonGroup } from './GoalApplicationButtonGroup';
+
+const mockGoal = {
+  netMonthlySalary: 6000,
+  taxesPercentage: 0.2,
+  rothContributionPercentage: 0.1,
+  traditionalContributionPercentage: 0.04,
+  ministryExpenses: {
+    benefitsCharge: 2000,
+    ministryMileage: 300,
+    medicalMileage: 50,
+    medicalExpenses: 300,
+    ministryPartnerDevelopment: 200,
+    communications: 100,
+    entertainment: 200,
+    staffDevelopment: 50,
+    supplies: 50,
+    technology: 50,
+    travel: 200,
+    transfers: 100,
+    other: 200,
+  },
+};
+
+describe('GoalApplicationButtonGroup', () => {
+  it('renders both buttons initially', () => {
+    const { getByRole } = render(
+      <GoalCalculatorTestWrapper>
+        <GoalApplicationButtonGroup goal={mockGoal} />
+      </GoalCalculatorTestWrapper>
+    );
+
+    expect(
+      getByRole('button', { name: /finish & apply goal/i })
+    ).toBeInTheDocument();
+    expect(
+      getByRole('button', { name: /save goal without applying/i })
+    ).toBeInTheDocument();
+  });
+
+  it('hides both buttons when "Save Goal Without Applying" is clicked', () => {
+    const { queryByRole, getByRole } = render(
+      <GoalCalculatorTestWrapper>
+        <GoalApplicationButtonGroup goal={mockGoal} />
+      </GoalCalculatorTestWrapper>
+    );
+
+    const saveWithoutApplyingButton = getByRole('button', {
+      name: /save goal without applying/i,
+    });
+    userEvent.click(saveWithoutApplyingButton);
+    expect(
+      queryByRole('button', { name: /finish & apply goal/i })
+    ).not.toBeInTheDocument();
+    expect(saveWithoutApplyingButton).not.toBeInTheDocument();
+  });
+
+  it('shows loading state when "Finish & Apply Goal" is clicked', async () => {
+    const { getByRole } = render(
+      <GoalCalculatorTestWrapper>
+        <GoalApplicationButtonGroup goal={mockGoal} />
+      </GoalCalculatorTestWrapper>
+    );
+
+    const finishButton = getByRole('button', {
+      name: /finish & apply goal/i,
+    });
+    userEvent.click(finishButton);
+    expect(getByRole('button', { name: /saving.../i })).toBeInTheDocument();
+    expect(getByRole('progressbar')).toBeInTheDocument();
+    expect(finishButton).toBeDisabled();
+  });
+
+  it('shows success message when finish button is clicked', async () => {
```

Comment:
Can you make this test check that it makes the GraphQL mutation? I would also merge it with the test above. Then we have one test that tests everything that should happen when the user clicks the "Finish" button.